src/jdk.jpackage/share/classes/jdk/jpackage/internal/IOUtils.java

```diff
@@ -83,10 +90,17 @@ public FileVisitResult preVisitDirectory(Path dir,
             @Override
             public FileVisitResult postVisitDirectory(Path dir, IOException e)
                             throws IOException {
-                Files.delete(dir);
+                try {
+                    Files.delete(dir);
+                } catch (IOException ex) {
+                    exception.compareAndSet(ex, null);
```

Comment:
The arguments are backwards. The first argument is the one used for comparison, and if the current reference is equal to the first, it is set to the second value.